test/suites/integration/https_listener_weighted_rule_with_service_export_import_test.go

```diff
@@ -99,55 +108,56 @@ var _ = Describe("Test 2 listeners gateway with weighted httproute rules and ser
 					nonDefaultRule := lo.Filter(listRulesResp.Items, func(rule *vpclattice.RuleSummary, _ int) bool {
 						return *rule.IsDefault == false
 					})
-					Expect(nonDefaultRule).To(HaveLen(1))
+					g.Expect(nonDefaultRule).To(HaveLen(1))
 					retrievedWeightedTGRule, _ := testFramework.LatticeClient.GetRuleWithContext(ctx, &vpclattice.GetRuleInput{
 						ServiceIdentifier:  vpcLatticeService.Id,
 						ListenerIdentifier: listenerId,
 						RuleIdentifier:     nonDefaultRule[0].Id,
 					})
 					retrievedWeightedTargetGroup0InRule := retrievedWeightedTGRule.Action.Forward.TargetGroups[0]
 					retrievedWeightedTargetGroup1InRule := retrievedWeightedTGRule.Action.Forward.TargetGroups[1]
-					Expect(*retrievedWeightedTargetGroup0InRule.TargetGroupIdentifier).To(Equal(*retrievedTg0.Id))
-					Expect(*retrievedWeightedTargetGroup0InRule.Weight).To(BeEquivalentTo(20))
-					Expect(*retrievedWeightedTargetGroup1InRule.TargetGroupIdentifier).To(Equal(*retrievedTg1.Id))
-					Expect(*retrievedWeightedTargetGroup1InRule.Weight).To(BeEquivalentTo(80))
+					g.Expect(*retrievedWeightedTargetGroup0InRule.TargetGroupIdentifier).To(Equal(*retrievedTgs[0].Id))
+					g.Expect(*retrievedWeightedTargetGroup0InRule.Weight).To(BeEquivalentTo(20))
+					g.Expect(*retrievedWeightedTargetGroup1InRule.TargetGroupIdentifier).To(Equal(*retrievedTgs[1].Id))
+					g.Expect(*retrievedWeightedTargetGroup1InRule.Weight).To(BeEquivalentTo(80))
 				}
 			}).Should(Succeed())
-			log.Println("Verifying Weighted rule traffic")
-			dnsName := testFramework.GetVpcLatticeServiceDns(httpRoute.Name, httpRoute.Namespace)
-
-			pods := testFramework.GetPodsByDeploymentName(deployment0.Name, deployment0.Namespace)
-			Expect(len(pods)).To(BeEquivalentTo(1))
-			log.Println("client pod name:", pods[0].Name)
-			protocols := []string{"http", "https"}
-			for _, protocol := range protocols {
+			Eventually(func(g Gomega) { // Put traffic verification logic in the Eventually() block, because connectivity setup may need some time to fully propagate to vpc lattice dataplane
+				log.Println("Verifying Weighted rule traffic")
+				dnsName := testFramework.GetVpcLatticeServiceDns(httpRoute.Name, httpRoute.Namespace)
```

Comment:
Just move the verification logic in the `Eventually()` block, verification logic itself didn't change